pkg/operator/ceph/object/rgw.go

```diff
@@ -91,55 +116,160 @@ func (c *clusterConfig) createOrUpdate(update bool) error {
 }
 
 func (c *clusterConfig) startRGWPods(update bool) error {
+	// backward compatibility, triggered during updates
+	if c.store.Spec.Gateway.AllNodes {
+		//  log info we don't support that anymore
+		logger.Infof("setting AllNodes to %t is not supported anymore, removing old DaemonSets if any and replace them with an equivalent number of Deployments in object store %s", c.store.Spec.Gateway.AllNodes, c.store.Name)
 
-	// if intended to update, remove the old pods so they can be created with the new spec settings
-	if update {
-		err := k8sutil.DeleteDeployment(c.context.Clientset, c.store.Namespace, c.instanceName())
+		// Make a best effort to delete the rgw pods daemonsets
+		storeLabelSelector := fmt.Sprintf("rook_object_store=%s", c.store.Name)
+		daemons, err := k8sutil.GetDaemonsets(c.context.Clientset, c.store.Namespace, storeLabelSelector)
 		if err != nil {
-			logger.Warning(err.Error())
+			logger.Warningf("could not get deployments for object store %s (matching label selector '%s'): %+v", c.store.Name, storeLabelSelector, err)
 		}
-		err = k8sutil.DeleteDaemonset(c.context.Clientset, c.store.Namespace, c.instanceName())
-		if err != nil {
-			logger.Warning(err.Error())
+		daemonsetNum := len(daemons.Items)
```

Comment:
I've thought about that but I haven't found anything to get the list of nodes. happy to do it if you have any tip. Thanks.

Reply:
Maybe it's not worth worrying about. What if we just allow the code to continue using `Instances`? Then they will be motivated to update the `Instances` if there aren't enough.

Reply:
Well currently, we count the number of daemonsets and use that as a reference to bootstrap deployments. The only downside is that users must not forget to update their spec afterwards but that seems smoother.

Reply:
The problem is if the operator is restarted after the daemonset is deleted, the operator will just use `instances` since it doesn't have the number of rgw pods in the daemonset anymore. So we can't reliably use the pod count and be idempotent.

Reply:
I'm not sure to follow, "if the operator gets restarted after the daemonset is deleted" --> so someone restarted this? It's kind of bad timing.
To me, this sounds acceptable and seems more like a corner case. Assuming this happens, it's not so different from https://github.com/rook/rook/issues/3245 and no rgws are running anymore. At this point, the user simply needs to update its `instances` count to whatever they want.